lib/alerts/medicine_article_monitor.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+# This class to create alerts when a WikiProject Medicine article
+# is assigned for a course that doesn't have the medical content
+# training module.
+# wiki_education mode only
+# Goal: to catch cases where a class that wasn't expected
+# to work on medical topics tries to do so.
+# Issue #4693
+class MedicineArticleMonitor
+  WIKI_PROJECT_MEDICINE_CAT = 'All_WikiProject_Medicine_articles'
+  MEDICAL_TRAINING_IDS = [11, 48].freeze
+
+  def self.create_alerts_for_no_med_training_for_course
+    monitor = new
+    monitor.refresh_med_article_titles
+    monitor.create_alerts_for_no_med_training_for_course
+  end
+
+  def initialize
+    @assignment_time_for_alert = 1.day.ago
+    @assignment_time_for_refresh_titles = 1.day.ago
+    @med_category = med_category
+  end
+
+  def create_alerts_for_no_med_training_for_course
+    last_assigned_med_articles.each do |assignment|
+      unless med_training_for_course?(assignment.course_id)
```

Comment:
One inefficiency here is that it will run the `med_training_for_course?` query once per Assignment, even if all the assignments are for the same course.

It think this is okay; it's probably not a significant bottleneck, and the code is clear and isolated to this file. But if we wanted to make this a little faster, we could include the `Course` in the `last_assigned_articles` query, and make `med_training_for_course` into an instance method of `Course` (with its value cached in an instance variable the first time it gets checked). (Other strategies to work around this N+1 situation without changing `Course` are also possible.)

Reply:
I'm not requesting any changes to this (just thinking it through in writing).